File: src/main/java/br/com/blz/testjava/gateways/http/response/ProductResponseDTO.java
package br.com.blz.testjava.gateways.http.response;

import lombok.Builder;
import lombok.Data;

@Builder
@Data
public class ProductResponseDTO {

    private Long id;
    private String name;
    private Integer sku;
    private boolean isMarketable;
    private InventoryResponseDTO inventory;
}

File: src/main/java/br/com/blz/testjava/product/dto/ProductResponseDTO.kt
package br.com.blz.testjava.product.dto

import br.com.blz.testjava.types.WarehouseType

data class ProductResponseDTO(
  val sku: Long,
  val name: String,
  val inventory: ProductInventoryResponseDTO,
  val isMarketable: Boolean
)

data class ProductInventoryResponseDTO(
  val quantity: Int,
  val warehouses: List<WareHouseResponseDTO>
)

data class WareHouseResponseDTO(
  val locality: String,
  val quantity: Int,
  val type: WarehouseType
)

